mysticeti-core/src/commit_observer.rs

```diff
@@ -180,3 +180,47 @@ impl<H: ProcessedTransactionHandler<TransactionLocator> + Send + Sync> CommitObs
         self.commit_interpreter.committed = committed;
     }
 }
+
+pub struct SimpleCommitObserver {
+    commit_interpreter: Linearizer,
+    /// A channel to send committed sub-dags to the consumer of consensus output.
+    sender: tokio::sync::mpsc::Sender<CommittedSubDag>,
+}
+
+impl SimpleCommitObserver {
+    pub fn new(sender: tokio::sync::mpsc::Sender<CommittedSubDag>) -> Self {
+        Self {
+            commit_interpreter: Linearizer::new(),
+            sender,
+        }
+    }
+}
+
+impl CommitObserver for SimpleCommitObserver {
+    fn handle_commit(
+        &mut self,
+        block_store: &BlockStore,
+        committed_leaders: Vec<Data<StatementBlock>>,
+    ) -> Vec<CommittedSubDag> {
+        let committed = self
+            .commit_interpreter
+            .handle_commit(block_store, committed_leaders);
+        for commit in &committed {
+            // TODO: Could we get rid of this clone latter?
+            self.sender.send(commit.clone()).unwrap();
```

Comment:
I thought we would want that? In general if the receiver stopped unexpectedly, there is no point to keep sending and mysticeti should panic.

Reply:
If mysticeti panics, it will bring down whole process. I am not super opposed to `unwrap` here, just need to keep in mind that now we must stop and wait for mysticeti to stop first, before stopping the task that polls this channel